Pod restarted after PipelineRun deleted.

# Expected behavior
Show status of PipelineRun

# Actual behavior
Pod restarted

# Steps to reproduce the problem

1. kubectl -n <namespace> delete -f <pipeline_run>

# Environment
- Kubernetes Platform: kubernetes cloud working on openstack.

- Tekton Pipelines version:
v0.11.0-rc4

- Tekton Dashboard version:
v0.6.0

# Additional Info
```
{"level":"debug","msg":"Secret Controller detected secret 'git-ssh' deleted"}
{"level":"debug","msg":"Secret Controller detected secret 'robot-git-ssh-token-gpqlt' deleted"}
{"level":"debug","msg":"PipelineResource Controller detected pipelineResource 'pipeline-git-repo' deleted"}
{"level":"debug","msg":"Task Controller detected task 'list-pipeline-repo-files' deleted"}
{"level":"debug","msg":"Task Controller detected task 'build-pipeline-repo-files' deleted"}
{"level":"debug","msg":"Pipeline Controller detected pipeline 'pipeline-list-pipeline-repo-files' deleted"}
{"level":"debug","msg":"PipelineRun Controller detected pipelineRun 'demo-condtional-pr-without-condition-resource' deleted"}
E0402 09:15:51.231931       1 runtime.go:78] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x13c50c0), concrete:(*runtime._type)(0x145f300), asserted:(*runtime._type)(0x1587460), missingMethod:""} (interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.ServiceAccount)
goroutine 178 [running]:
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1400080, 0xc0000b3620)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1400080, 0xc0000b3620)
	runtime/panic.go:679 +0x1b2
github.com/tektoncd/dashboard/pkg/controllers/kubernetes.serviceAccountDeleted(0x145f300, 0xc0007485e0)
	github.com/tektoncd/dashboard/pkg/controllers/kubernetes/serviceAccount.go:61 +0x157
github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnDelete(...)
	github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache/controller.go:212
github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0x1, 0x0, 0x0)
	github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache/shared_informer.go:660 +0x1d6
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc00053ddd8, 0x0, 0xc0003a4eb8)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:292 +0x51
github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache/shared_informer.go:652 +0x79
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00018f740)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00053df40, 0xdf8475800, 0x0, 0x42cf01, 0xc0005650e0)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache.(*processorListener).run(0xc000532180)
	github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache/shared_informer.go:650 +0x9b
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc00051c1b0, 0xc000534f40)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x59
created by github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:69 +0x62
panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.ServiceAccount [recovered]
	panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.ServiceAccount

goroutine 178 [running]:
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1400080, 0xc0000b3620)
	runtime/panic.go:679 +0x1b2
github.com/tektoncd/dashboard/pkg/controllers/kubernetes.serviceAccountDeleted(0x145f300, 0xc0007485e0)
	github.com/tektoncd/dashboard/pkg/controllers/kubernetes/serviceAccount.go:61 +0x157
github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnDelete(...)
	github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache/controller.go:212
github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0x1, 0x0, 0x0)
	github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache/shared_informer.go:660 +0x1d6
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc0007f7dd8, 0x0, 0xc0003a4eb8)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:292 +0x51
github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache/shared_informer.go:652 +0x79
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00018f740)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0007f7f40, 0xdf8475800, 0x0, 0x42cf01, 0xc0005650e0)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache.(*processorListener).run(0xc000532180)
	github.com/tektoncd/dashboard/vendor/k8s.io/client-go/tools/cache/shared_informer.go:650 +0x9b
github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc00051c1b0, 0xc000534f40)
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x59
created by github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	github.com/tektoncd/dashboard/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:69 +0x62
```

--- a/pkg/controllers/kubernetes/extension.go
+++ b/pkg/controllers/kubernetes/extension.go
@@ -2,6 +2,7 @@ package kubernetes
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	"github.com/tektoncd/dashboard/pkg/router"
@@ -95,11 +96,13 @@ func (e extensionHandler) serviceUpdated(oldObj, newObj interface{}) {
 }
 
 func (e extensionHandler) serviceDeleted(obj interface{}) {
-	service := obj.(*v1.Service)
-	if service.Namespace == e.installNamespace {
-		if value := service.Labels[router.ExtensionLabelKey]; value == router.ExtensionLabelValue {
-			logging.Log.Debugf("Extension Controller detected extension '%s' deleted", service.Name)
-			e.UnregisterExtension(service)
+	serviceMeta := utils.GetDeletedObjectMeta(obj)
+	if serviceMeta.GetNamespace() == e.installNamespace {
+		if value := serviceMeta.GetLabels()[router.ExtensionLabelKey]; value == router.ExtensionLabelValue {
+			logging.Log.Debugf("Extension Controller detected extension '%s' deleted", serviceMeta.GetName())
+			if serviceMeta.GetUID() != "" {
+				e.UnregisterExtensionByMeta(serviceMeta)
+			}
 			data := broadcaster.SocketData{
 				MessageType: broadcaster.ExtensionDeleted,
 				Payload:     obj,

--- a/pkg/controllers/kubernetes/namespaces.go
+++ b/pkg/controllers/kubernetes/namespaces.go
@@ -2,6 +2,7 @@ package kubernetes
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1 "k8s.io/api/core/v1"
@@ -31,7 +32,7 @@ func namespaceCreated(obj interface{}) {
 }
 
 func namespaceDeleted(obj interface{}) {
-	logging.Log.Debugf("Namespace Controller detected namespace '%s' deleted", obj.(*v1.Namespace).Name)
+	logging.Log.Debugf("Namespace Controller detected namespace '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.NamespaceDeleted,
 		Payload:     obj,

--- a/pkg/controllers/kubernetes/secret.go
+++ b/pkg/controllers/kubernetes/secret.go
@@ -2,6 +2,7 @@ package kubernetes
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1 "k8s.io/api/core/v1"
@@ -45,7 +46,7 @@ func secretUpdated(oldObj, newObj interface{}) {
 }
 
 func secretDeleted(obj interface{}) {
-	logging.Log.Debugf("Secret Controller detected secret '%s' deleted", obj.(*v1.Secret).Name)
+	logging.Log.Debugf("Secret Controller detected secret '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.SecretDeleted,
 		Payload:     obj,

--- a/pkg/controllers/kubernetes/serviceAccount.go
+++ b/pkg/controllers/kubernetes/serviceAccount.go
@@ -15,6 +15,7 @@ package kubernetes
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1 "k8s.io/api/core/v1"
@@ -58,7 +59,7 @@ func serviceAccountUpdated(oldObj, newObj interface{}) {
 }
 
 func serviceAccountDeleted(obj interface{}) {
-	logging.Log.Debugf("ServiceAccount Controller detected ServiceAccount '%s' deleted", obj.(*v1.ServiceAccount).Name)
+	logging.Log.Debugf("ServiceAccount Controller detected ServiceAccount '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.ServiceAccountDeleted,
 		Payload:     obj,

--- a/pkg/controllers/tekton/clustertask.go
+++ b/pkg/controllers/tekton/clustertask.go
@@ -15,6 +15,7 @@ package tekton
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
@@ -55,7 +56,7 @@ func clusterTaskUpdated(oldObj, newObj interface{}) {
 }
 
 func clusterTaskDeleted(obj interface{}) {
-	logging.Log.Debugf("Cluster Task Controller detected clusterTask '%s' deleted", obj.(*v1alpha1.ClusterTask).Name)
+	logging.Log.Debugf("Cluster Task Controller detected clusterTask '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.ClusterTaskDeleted,
 		Payload:     obj,

--- a/pkg/controllers/tekton/pipeline.go
+++ b/pkg/controllers/tekton/pipeline.go
@@ -2,6 +2,7 @@ package tekton
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
@@ -43,7 +44,7 @@ func pipelineUpdated(oldObj, newObj interface{}) {
 }
 
 func pipelineDeleted(obj interface{}) {
-	logging.Log.Debugf("Pipeline Controller detected pipeline '%s' deleted", obj.(*v1alpha1.Pipeline).Name)
+	logging.Log.Debugf("Pipeline Controller detected pipeline '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.PipelineDeleted,
 		Payload:     obj,

--- a/pkg/controllers/tekton/pipelineresource.go
+++ b/pkg/controllers/tekton/pipelineresource.go
@@ -2,6 +2,7 @@ package tekton
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
@@ -43,7 +44,7 @@ func pipelineResourceUpdated(oldObj, newObj interface{}) {
 }
 
 func pipelineResourceDeleted(obj interface{}) {
-	logging.Log.Debugf("PipelineResource Controller detected pipelineResource '%s' deleted", obj.(*v1alpha1.PipelineResource).Name)
+	logging.Log.Debugf("PipelineResource Controller detected pipelineResource '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.PipelineResourceDeleted,
 		Payload:     obj,

--- a/pkg/controllers/tekton/pipelinerun.go
+++ b/pkg/controllers/tekton/pipelinerun.go
@@ -2,6 +2,7 @@ package tekton
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
@@ -43,7 +44,7 @@ func pipelineRunUpdated(oldObj, newObj interface{}) {
 }
 
 func pipelineRunDeleted(obj interface{}) {
-	logging.Log.Debugf("PipelineRun Controller detected pipelineRun '%s' deleted", obj.(*v1alpha1.PipelineRun).Name)
+	logging.Log.Debugf("PipelineRun Controller detected pipelineRun '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.PipelineRunDeleted,
 		Payload:     obj,

--- a/pkg/controllers/tekton/task.go
+++ b/pkg/controllers/tekton/task.go
@@ -2,6 +2,7 @@ package tekton
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
@@ -42,7 +43,7 @@ func taskUpdated(oldObj, newObj interface{}) {
 }
 
 func taskDeleted(obj interface{}) {
-	logging.Log.Debugf("Task Controller detected task '%s' deleted", obj.(*v1alpha1.Task).Name)
+	logging.Log.Debugf("Task Controller detected task '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.TaskDeleted,
 		Payload:     obj,

--- a/pkg/controllers/tekton/taskrun.go
+++ b/pkg/controllers/tekton/taskrun.go
@@ -2,6 +2,7 @@ package tekton
 
 import (
 	"github.com/tektoncd/dashboard/pkg/broadcaster"
+	"github.com/tektoncd/dashboard/pkg/controllers/utils"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
@@ -42,7 +43,7 @@ func taskRunUpdated(oldObj, newObj interface{}) {
 }
 
 func taskRunDeleted(obj interface{}) {
-	logging.Log.Debugf("TaskRun Controller detected taskRun '%s' deleted", obj.(*v1alpha1.TaskRun).Name)
+	logging.Log.Debugf("TaskRun Controller detected taskRun '%s' deleted", utils.GetDeletedObjectMeta(obj).GetName())
 	data := broadcaster.SocketData{
 		MessageType: broadcaster.TaskRunDeleted,
 		Payload:     obj,

--- a/pkg/controllers/utils/utils.go
+++ b/pkg/controllers/utils/utils.go
@@ -25,8 +25,7 @@ func GetDeletedObjectMeta(obj interface{}) metav1.Object {
 	// Deal with tombstone events by pulling the object out.  Tombstone events wrap the object in a
 	// DeleteFinalStateUnknown struct, so the object needs to be pulled out.
 	// Copied from sample-controller
-	// This should never happen if we aren't missing events, which we have concluded that we are not
-	// and made decisions off of this belief.  Maybe this shouldn't be here?
+	// This should only happen when we're missing events.
 	if _, ok := obj.(metav1.Object); !ok {
 		// If the object doesn't have Metadata, assume it is a tombstone object of type DeletedFinalStateUnknown
 		if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); !ok {

--- a/pkg/controllers/utils/utils.go
+++ b/pkg/controllers/utils/utils.go
@@ -1,0 +1,48 @@
+/*
+Copyright 2020 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	logging "github.com/tektoncd/dashboard/pkg/logging"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Adapted from https://github.com/kubernetes-sigs/controller-runtime/blob/v0.5.2/pkg/source/internal/eventsource.go#L131-L149
+func GetDeletedObjectMeta(obj interface{}) metav1.Object {
+	// Deal with tombstone events by pulling the object out.  Tombstone events wrap the object in a
+	// DeleteFinalStateUnknown struct, so the object needs to be pulled out.
+	// Copied from sample-controller
+	// This should never happen if we aren't missing events, which we have concluded that we are not
+	// and made decisions off of this belief.  Maybe this shouldn't be here?
+	if _, ok := obj.(metav1.Object); !ok {
+		// If the object doesn't have Metadata, assume it is a tombstone object of type DeletedFinalStateUnknown
+		if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); !ok {
+			logging.Log.Errorf("Error decoding object: Expected cache.DeletedFinalStateUnknown, got %T", obj)
+			return &metav1.ObjectMeta{}
+		} else {
+			// Set obj to the tombstone obj
+			obj = tombstone.Obj
+		}
+	}
+
+	// Pull metav1.Object out of the object
+	if o, err := meta.Accessor(obj); err != nil {
+		logging.Log.Errorf("Missing meta for object %T: %v", obj, err)
+		return &metav1.ObjectMeta{}
+	} else {
+		return o
+	}
+}

--- a/pkg/router/router.go
+++ b/pkg/router/router.go
@@ -29,6 +29,7 @@ import (
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ExtensionLabelKey is the label key required by services to be registered as a
@@ -120,12 +121,18 @@ func (h *Handler) RegisterExtension(extensionService *corev1.Service) {
 // UnregisterExtension unregisters an extension. This should be called BEFORE
 // registration of extensionService on informer update
 func (h *Handler) UnregisterExtension(extensionService *corev1.Service) {
-	logging.Log.Infof("Removing extension %s", extensionService.Name)
+	h.UnregisterExtensionByMeta(&extensionService.ObjectMeta)
+}
+
+// UnregisterExtensionByMeta unregisters an extension by its metadata. This
+// should be called BEFORE registration of extensionService on informer update
+func (h *Handler) UnregisterExtensionByMeta(extensionService metav1.Object) {
+	logging.Log.Infof("Removing extension %s", extensionService.GetName())
 	h.Lock()
 	defer h.Unlock()
 
 	// Grab endpoints to remove from service
-	ext := h.uidExtensionMap[string(extensionService.UID)]
+	ext := h.uidExtensionMap[string(extensionService.GetUID())]
 	for _, path := range ext.endpoints {
 		extensionPath := extensionPath(ext.Name, path)
 		fullPath := fmt.Sprintf("%s/%s", h.extensionWebService.RootPath(), extensionPath)
@@ -139,7 +146,7 @@ func (h *Handler) UnregisterExtension(extensionService *corev1.Service) {
 		h.extensionWebService.RemoveRoute(fullPath+"/{var:*}", "PUT")
 		h.extensionWebService.RemoveRoute(fullPath+"/{var:*}", "DELETE")
 	}
-	delete(h.uidExtensionMap, string(extensionService.UID))
+	delete(h.uidExtensionMap, string(extensionService.GetUID()))
 }
 
 // registerExtensions registers the WebService responsible for

